packages/debugger/app/page.tsx

```diff
@@ -36,19 +42,68 @@ export default function Page({
     likedCast: false,
     recastedCast: false,
   });
+  const currentChainId = useChainId();
+  const config = useConfig();
+  const account = useAccount();
+  const { openConnectModal } = useConnectModal();
+
   useEffect(() => {
     if (url !== urlInput && url) {
       setUrlInput(url);
     }
   }, [url]);
   const signerState = useFarcasterIdentity();
+
+  const onTransaction: OnTransactionFunc = useCallback(
+    async ({ transactionData }) => {
+      const { params, chainId, method } = transactionData;
+      if (!chainId.startsWith("eip155:")) {
+        alert(`debugger: Unrecognized chainId ${chainId}`);
+        return null;
+      }
+
+      if (!account.address) {
+        openConnectModal?.();
+        return null;
+      }
+
+      const requestedChainId = parseInt(chainId.split("eip155:")[1]!);
+
+      if (currentChainId !== requestedChainId) {
+        console.log("switching chain");
+        await switchChain(config, {
+          chainId: requestedChainId,
+        });
+      }
+
+      try {
+        // Send the transaction
+        console.log("sending tx");
+        const transactionId = await sendTransaction(config, {
+          to: params.to,
+          data: params.data,
+          value: BigInt(params.value),
+        });
+        return transactionId;
+      } catch (error) {
+        console.error(error);
+        return null;
+      }
+    },
+    []
```

Comment:
this function keeps logging no connected wallet because it is missing dependencies @stephancill

Reply:
ahh thank you for pointing that out - i think that fixes the inconsistent behaviour i was experiencing